components/test_raftstore/src/transport_simulate.rs

```diff
@@ -600,7 +603,11 @@ impl Filter<StoreMsg> for LeadingDuplicatedSnapshotFilter {
                     if msg.get_message().get_msg_type() == MessageType::MsgSnapshot && !stale {
                         if last_msg.as_ref().map_or(false, |l| l != &msg) {
                             to_send.push(StoreMsg::RaftMessage(last_msg.take().unwrap()));
-                            *last_msg = Some(msg);
+                            if self.together {
+                                to_send.push(StoreMsg::RaftMessage(msg));
```

Comment:
So if the message sequence is `snapshot[1], snapshot[2], snapshot[2], snapshot[2]`, after applying the filter, the sequence will not be changed?

Reply:
yes, this filter is just to ensure two different snapshots show up together once. It doesn't care the rest messages.